homework-submission/myapp/src/stores/rashadStore.js

```diff
@@ -0,0 +1,80 @@
+import { configure, action, observable, runInAction, computed } from "mobx";
+
+configure({ enforceActions: true });
+
+const url = "https://hyf-react-api.herokuapp.com";
+
+class RashadStore {
+  @observable todosList = [];
+
+  @action
+  checker = item => {
+    item.done = !item.done;
+    fetch(`${url}/todos/${item._id}`, {
+      method: "PATCH",
+      body: JSON.stringify({ done: item.done }),
+      headers: { "content-type": "application/json" }
+    })
+      .then(response => response.json())
+      .then(todo => {
+        const newTodos = this.todosList.map(element => {
+          if (todo._id === element._id) {
+            return { ...element, done: !element.done };
+          }
+          return element;
+        });
+        runInAction(() => {
+          this.todosList = newTodos;
+        });
+      })
+      .catch(err => console.log(err));
+  };
+
+  @action
+  remover = item => {
+    fetch(`${url}/todos/${item._id}`, {
+      method: "DELETE",
+      headers: { "content-type": "application/json" }
+    })
+      .then(response => response.json())
+      .then(todo => {
+        const newTodos = this.todosList.filter(
+          element => element._id !== todo._id
+        );
+        runInAction(() => {
+          this.todosList = newTodos;
+        });
+      })
+      .catch(err => console.log(err));
+  };
+
+  @action
+  submitHandler = e => {
+    e.preventDefault();
```

Comment:
Keep the event (e) and any view related code on the component itself and just pass the values/object to your function directly.